src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs

```diff
@@ -522,6 +523,47 @@ public void HandleLogMessage(TestMessageLevel level, string message)
         /// <param name="rawMessage"></param>
         public void HandleRawMessage(string rawMessage)
         {
+            if (this.requestData.IsTelemetryOptedIn)
+            {
+                var message = this.dataSerializer.DeserializeMessage(rawMessage);
+
+                if (string.Equals(message.MessageType, MessageType.ExecutionComplete))
+                {
+                    var testRunCompletePayload =
+                        this.dataSerializer.DeserializePayload<TestRunCompletePayload>(message);
+
+                    if (testRunCompletePayload != null)
+                    {
+                        if (testRunCompletePayload.TestRunCompleteArgs?.Metrics == null)
+                        {
+                            testRunCompletePayload.TestRunCompleteArgs.Metrics = new Dictionary<string, object>();
+                        }
+
+                        foreach (var kvp in this.requestData.MetricsCollection.Metrics)
+                        {
+                            testRunCompletePayload.TestRunCompleteArgs.Metrics.Add(kvp.Key, kvp.Value);
+                        }
+
+                        var executionTotalTimeTakenForDesignMode = DateTime.UtcNow - this.executionStartTime;
+
+                        // Fill in the time taken to complete the run
+                        testRunCompletePayload.TestRunCompleteArgs.Metrics[TelemetryDataConstants.TimeTakenInSecForRun] = executionTotalTimeTakenForDesignMode.TotalSeconds;
+                    }
```

Comment:
I was expecting other telemetry points like console parameters will also be added